mempool/src/error.rs

```diff
@@ -0,0 +1,81 @@
+use thiserror::Error;
+
+use common::chain::transaction::Transaction;
+use common::chain::OutPoint;
+use common::primitives::amount::Amount;
+use common::primitives::Id;
+use common::primitives::H256;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Mempool is full")]
+    MempoolFull,
+    #[error(transparent)]
+    TxValidationError(TxValidationError),
+}
+
+#[derive(Debug, Error)]
+pub enum TxValidationError {
+    #[error("No Inputs")]
+    NoInputs,
+    #[error("No Ouputs")]
+    NoOutputs,
+    #[error("DuplicateInputs")]
+    DuplicateInputs,
+    #[error("LooseCoinbase")]
+    LooseCoinbase,
+    #[error("OutPointNotFound {outpoint:?}")]
+    OutPointNotFound {
+        outpoint: OutPoint,
+        tx_id: Id<Transaction>,
+    },
+    #[error("ExceedsMaxBlockSize")]
+    ExceedsMaxBlockSize,
+    #[error("TransactionAlreadyInMempool")]
+    TransactionAlreadyInMempool,
+    #[error("ConflictWithIrreplaceableTransaction")]
```

Comment:
Nitpick: as far as I can see, we mostly use a more "human readable" description of errors. So it probably should be "Conflict with irreplaceable transaction" instead. Same for other errors.

Reply:
Fixed, thanks.